dts/bindings/clock/nordic,nrf-hsfll.yaml

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic nRF HSFLL
+
+  The HSFLL mixed-mode IP generates several clock frequencies in the range from
+  64 MHz to 400 MHz (in steps of 16 MHz).
+
+  Usage example:
+
+    hsfll: clock@deadbeef {
+        compatible = "nordic,nrf-hsfll";
+        reg = <0xdeadbeef 0x1000>;
+        clocks = <&fll16m>;
+        clock-frequency = <DT_FREQ_M(320)>;
+        nordic,ficrs = <&ficr NRF_FICR_TRIM_APPLICATION_HSFLL_TRIM_VSUP>,
+                       <&ficr NRF_FICR_TRIM_APPLICATION_HSFLL_TRIM_COARSE_0>,
+                       <&ficr NRF_FICR_TRIM_APPLICATION_HSFLL_TRIM_FINE_0>;
+        nordic,ficr-names = "vsup", "coarse", "fine";
+    };
+
+  Required FICR entries are for VSUP, COARSE and FINE trim values.
+
+compatible: "nordic,nrf-hsfll"
+
+include: [base.yaml, fixed-clock.yaml, nordic-nrf-ficr-client.yaml]
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    required: true
+
+  clock-frequency:
+    enum:
+      - 64000000
+      - 80000000
+      - 96000000
+      - 112000000
+      - 128000000
+      - 144000000
+      - 160000000
+      - 176000000
+      - 192000000
+      - 208000000
+      - 224000000
+      - 240000000
+      - 256000000
+      - 272000000
+      - 288000000
+      - 304000000
+      - 320000000
+      - 336000000
+      - 352000000
+      - 368000000
+      - 384000000
+      - 400000000
+
+  nordic,ficrs:
+    required: true
+
+  nordic,ficr-names:
+    required: true
```

Comment:
Couldn't these properties be set as required in `nordic-nrf-ficr-client.yaml`?

Reply:
hmm what if some clients want them optional (e.g. conditioned to another property being set)?

Reply:
I just thought that it's a bit strange if a FICR client does not define at least the `nordic,ficrs` property. But let's keep it as it is for now.

Reply:
So we could have a binding that only required FICR entries under certain conditions. For example, imagine we have a PLL with a `nordic,mode` property. If `mode` is "open", FICR is not read by the driver, whereas if "closed", some FICR tune values are needed. I'm not sure if a case like that will ever happen, but I assume it's not impossible that's why I haven't made the requirement at client YAML file. It's relatively similar to `pinctrl-device.yaml`.

Reply:
Note that our binding language doesn't allow us to do fancy things like "required if property X is present", and added to the fact that required cannot be undone, we must play safe here.

Reply:
AFAICT from spec, FICR values are always going to be needed by these drivers for retrieving the production-defined default values to load into the HSFLL.TRIM registers, so I'm not sure if the case would happen either. But I understand the need for flexibility given the language constraints though, and I'm all for leaving that option open for us in the future so doing that here now makes sense.

My only suggestion then would be that if we do leave FICR things open as "optional", to maybe include some clarification in a binding's description text they're not optional from a user configuration standpoint (because they aren't pick-and-choose settings and are there for hardware operation), just for driver flexibility.